src/addons/shallowCompare.js

```diff
@@ -17,10 +17,11 @@ var shallowEqual = require('shallowEqual');
  * Does a shallow comparison for props and state.
  * See ReactComponentWithPureRenderMixin
  */
-function shallowCompare(instance, nextProps, nextState) {
+function shallowCompare(instance, nextProps, nextState, nextContext) {
   return (
     !shallowEqual(instance.props, nextProps) ||
-    !shallowEqual(instance.state, nextState)
+    !shallowEqual(instance.state, nextState) ||
+    (typeof nextContext !== 'undefined' && !shallowEqual(instance.context, nextContext))
```

Comment:
Yeah, I don't think `shallowEqual` is going to add a meaningful amount of execution time.  The undefined check adds bytes, which increases download size; it might be a wash.  If we were going to do that check, we should just use `nextContext !== undefined` instead of doing a typeof and string compare, both of which are slow.

But in the interest of simplicity, let's just remove that undefined check.